WinUIB2CApp/MSAL/MSALClientHelper.cs

```diff
@@ -0,0 +1,248 @@
+        /// <returns></returns>
+        public async Task<String> SignInUserAndAcquireAccessToken(string[] scopes, string extraclaims)
+        {
+            var existingUser = await FetchSignedInUserFromCache().ConfigureAwait(false);
+
+            try
+            {
+                // Send the user to Azure AD for re-authentication as a silent acquisition wont resolve any CAE scenarios
+                this.AuthResult = await PublicClientApplication.AcquireTokenInteractive(scopes)
+                    .WithLoginHint(existingUser?.Username ?? String.Empty)
+                    .WithClaims(extraclaims)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+            }
+            catch (MsalException msalEx)
+            {
+                Debug.WriteLine($"Error Acquiring Token interactively:{Environment.NewLine}{msalEx}");
+            }
+
+            return this.AuthResult.AccessToken;
+        }
+
+        /// <summary>
+        /// Shows a pattern to sign-in a user interactively in applications that are input constrained and would need to fall-back on device code flow.
+        /// </summary>
+        /// <param name="scopes">The scopes.</param>
+        /// <param name="existingAccount">The existing account.</param>
+        /// <returns></returns>
+        private async Task<AuthenticationResult> SignInUserInteractivelyAsync(string[] scopes, IAccount existingAccount = null)
+        {
+            // If the operating system has UI
+            if (this.PublicClientApplication.IsUserInteractive())
+            {
+                return await this.PublicClientApplication.AcquireTokenInteractive(scopes)
+                    .WithLoginHint(existingAccount?.Username ?? String.Empty)
+                    .WithParentActivityOrWindow(windowHandle) // TODO: fix it
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+            }
+
+            // If the operating system does not have UI (e.g. SSH into Linux), you can fallback to device code, however this
+            // flow will not satisfy the "device is managed" CA policy.
+            return await this.PublicClientApplication.AcquireTokenWithDeviceCode(scopes, (dcr) =>
+            {
+                Console.WriteLine(dcr.Message);
+                return Task.CompletedTask;
+            }).ExecuteAsync().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Removes the first signed-in user's record from token cache
+        /// </summary>
+        public async void SignOutUser()
+        {
+            var existingUser = await FetchSignedInUserFromCache().ConfigureAwait(false);
+            this.SignOutUser(existingUser);
+        }
+
+        /// <summary>
+        /// Removes a given user's record from token cache
+        /// </summary>
+        /// <param name="user">The user.</param>
+        public async void SignOutUser(IAccount user)
```

Comment:
An account is not a user. A user can have several accounts in different tenants.